### PR TITLE
Internal: Support children creation handling using `createElementFromModel` for each element [ED-6217]

### PR DESCRIPTION
**Pull request by YitzhakAvm**

### CODE REVIEW

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/views/add-section/base.js [hunk 1]
```diff
@@ -119,7 +119,9 @@ class AddSectionBase extends Marionette.ItemView {
 			placeholder: false,
 			currentElementClass: 'elementor-html5dnd-current-element',
 			hasDraggingOnChildClass: 'elementor-dragging-on-child',
-			onDropping: this.onDropping.bind( this ),
+			getDropIndex: () => this.getOption( 'at' ),
+			// Merge different options if provided by child elements
+			...this.getDroppableOptions?.(),
```
[the remark this answers is not in the document]
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/add-section/base.js:124]: The purpose of utilizing the `createElementFromModel` method on all elements classes is to allow the modification of children creation behavior. `getDroppableOptions` will only affect the Droppable, which for itself is not much.

File: assets/dev/js/editor/views/add-section/inline.js [hunk 2]
```diff
@@ -35,10 +35,22 @@ class AddSectionView extends BaseAddSectionView {
...
 
+	getDroppableOptions() {
+		return {
+			onDropping: () => {
+				this.destroy();
+			},
+		};
+	}
+
 	onDropping() {
```
[the remark this answers is not in the document]
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/add-section/inline.js:46]: That's right, it should be deprecated in 3.6.0

File: assets/dev/js/editor/elements/views/column.js [hunk 3]
```diff
@@ -229,15 +230,11 @@ ColumnView = BaseElementView.extend( {
...
-			getDropContainer: () => this.getContainer(),
-			getDropIndex,
 			onDropping: ( side, event ) => {
-				event.stopPropagation();
-
-				// Triggering drag end manually, since it won't fired above iframe
-				elementor.getPreviewView().onPanelElementDragEnd();
-
-				this.addElementFromPanel( { at: getDropIndex( side, event ) } );
+				this.getContainer().view.onDropEvent(
```
Comment (StyleShit) [inline assets/dev/js/editor/elements/views/column.js:234]: Do you need that? You're already inside the view

```suggestion
				this.onDropEvent(
```

File: assets/dev/js/editor/views/add-section/base.js [hunk 4]
```diff
@@ -119,10 +119,22 @@ class AddSectionBase extends Marionette.ItemView {
 			placeholder: false,
 			currentElementClass: 'elementor-html5dnd-current-element',
 			hasDraggingOnChildClass: 'elementor-dragging-on-child',
-			onDropping: this.onDropping.bind( this ),
+			// Merge different options if provided by child elements
+			...this.getDroppableOptions?.(),
```
Comment (StyleShit) [inline assets/dev/js/editor/views/add-section/base.js:123]: Why do you need this `?.`, Isn't it always defined?

File: assets/dev/js/editor/views/add-section/inline.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/views/add-section/inline.js:47]: I still don't really get why it's here, let's talk about it offline?

File: assets/dev/js/editor/utils/jquery-html5-dnd.js [hunk 5]
```diff
@@ -332,36 +331,17 @@
...
-				} );
-			} else {
-				const dragged = elementor.channels.panelElements.request( 'element:selected' )?.model.attributes;
-
-				settings.getDropContainer().view.createElementFromModel(
-					{ elType: dragged.elType, widgetType: dragged.widgetType, custom: dragged.custom },
-					{
-						at: settings.getDropIndex( currentSide, event ),
-					}
-				);
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/jquery-html5-dnd.js:364]: 🔥🔥

File: assets/dev/js/editor/views/base-container.js [hunk 6]
```diff
@@ -155,6 +155,28 @@ module.exports = Marionette.CompositeView.extend( {
 		return widget;
 	},
 
+	onDropEvent( event, options ) {
```
Comment (StyleShit) [inline assets/dev/js/editor/views/base-container.js:158]: Why not just like this?
I mean, when you use the `onSomething` convention, others will usually assume it's an event already. What do you think?

```suggestion
	onDrop( event, options ) {
```

File: assets/dev/js/editor/views/base-container.js [hunk 7]
```diff
@@ -155,6 +155,28 @@ module.exports = Marionette.CompositeView.extend( {
...
+			$e.run( 'editor/browser-import/import', {
+				input,
+				target: this.getContainer(),
+				options: { event, target: { at: options.at } },
+			} );
+
+			return;
+		}
+
+		this.getContainer().view.createElementFromModel(
```
Comment (StyleShit) [inline assets/dev/js/editor/views/base-container.js:171]: Like above, do you need that here?

```suggestion
		this.createElementFromModel(
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/column.js (hunk 3 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/elements/views/column.js]: You're right. Fixed.

File: assets/dev/js/editor/views/add-section/base.js (hunk 4 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/add-section/base.js]: Previously this method was only existed in the `inline.js` file. Now when it's implemented here too, you're right, `?.` has no importance.
Fixed.

File: assets/dev/js/editor/views/base-container.js (hunk 6 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/base-container.js]: Done.

File: assets/dev/js/editor/views/base-container.js (hunk 7 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/base-container.js]: Done.

**Review by StyleShit — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please pull from develop

File: assets/dev/js/editor/elements/views/column.js [hunk 8]
```diff
@@ -247,6 +244,13 @@ ColumnView = BaseElementView.extend( {
 
 		this.addNewColumn();
 	},
+
+	createElementFromModel( ...args ) {
+		// Triggering drag end manually, since it won't fired above iframe
+		elementor.getPreviewView().onPanelElementDragEnd();
```
[the remark this answers is not in the document]
Comment (ItzhakAvm) [reply assets/dev/js/editor/elements/views/column.js:250]: No, it's supposed to run only for dnd events. It was taken from the original `onDropping` method of the `column`.

File: assets/dev/js/editor/utils/jquery-html5-dnd.js [hunk 9]
```diff
@@ -111,6 +109,7 @@
 
 		var initSettings = function() {
 			$.extend( settings, defaultSettings, userSettings );
+			self.settings = settings;
```
Comment (matipojo) [inline assets/dev/js/editor/utils/jquery-html5-dnd.js:112]: Where it's used?

File: assets/dev/js/editor/elements/views/column.js [hunk 10]
```diff
@@ -1,4 +1,5 @@
 import { DEFAULT_MAX_COLUMNS } from 'elementor-elements/views/section';
+import BaseSectionsContainerView from 'elementor-views/base-sections-container';
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/column.js:2]: Where it's used?

File: assets/dev/js/editor/elements/views/column.js [hunk 11]
```diff
@@ -231,6 +231,9 @@ ColumnView = BaseElementView.extend( {
 			placeholderClass: 'elementor-sortable-placeholder elementor-widget-placeholder',
 			hasDraggingOnChildClass: 'elementor-dragging-on-child',
 			onDropping: ( side, event ) => {
+				// Triggering drag end manually, since it won't fired above iframe
+				elementor.getPreviewView().onPanelElementDragEnd();
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/column.js:235]: It's relevant for column only?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/column.js (hunk 10 above)
Comment (ItzhakAvm) [inline assets/dev/js/editor/elements/views/column.js:2]:
```suggestion
```

Comment (ItzhakAvm) [reply assets/dev/js/editor/elements/views/column.js]: Removed.

File: assets/dev/js/editor/utils/jquery-html5-dnd.js (hunk 9 above)
Comment (ItzhakAvm) [inline assets/dev/js/editor/utils/jquery-html5-dnd.js:112]:
```suggestion
```

Comment (ItzhakAvm) [reply assets/dev/js/editor/utils/jquery-html5-dnd.js]: Apparently it's not anymore. Removed.

**Pull request merged**